Sources/TuistGenerator/Utils/CocoaPodsInteractor.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import TuistCore
+
+enum CocoaPodsInteractorError: FatalError, Equatable {
+    /// Thrown when CocoaPods cannot be found.
+    case cocoapodsNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .cocoapodsNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .cocoapodsNotFound:
+            return "CocoaPods was not found either in Bundler nor in the environment"
+        }
+    }
+}
+
+protocol CocoaPodsInteracting {
+    /// Runs 'pod install' for all the CocoaPods dependencies that have been indicated in the graph.
+    ///
+    /// - Parameter graph: Project graph.
+    /// - Throws: An error if the installation of the pods fails.
+    func install(graph: Graphing) throws
+}
+
+final class CocoaPodsInteractor: CocoaPodsInteracting {
+    /// Instance to output information to the user.
+    let printer: Printing
+
+    /// Instance to run commands in the system.
+    let system: Systeming
+
+    /// Initializes the CocoaPods
+    ///
+    /// - Parameters:
+    ///   - printer: Instance to output information to the user.
+    ///   - system: Instance to run commands in the system.
+    init(printer: Printing = Printer(),
+         system: Systeming = System()) {
+        self.printer = printer
+        self.system = system
+    }
+
+    /// Runs 'pod install' for all the CocoaPods dependencies that have been indicated in the graph.
+    ///
+    /// - Parameter graph: Project graph.
+    /// - Throws: An error if the installation of the pods fails.
+    func install(graph: Graphing) throws {
+        let canUseBundler = canUseCocoaPodsThroughBundler()
```

Comment:
worth adding a guard here to save a bit of time from running those shell calls if there aren't any cocoapod dependencies :)

```swift
guard !graph.cocoapods.isEmpty else { 
  return 
}
```